### PR TITLE
Prep for Cloud Conf

### LINKED ISSUE #94: Use Dependabot v2
I believe this would enable us to use [Dependabot to keep our workflows up-to-date](https://dependabot.com/github-actions/).

### LINKED ISSUE #97: Use BuildKit when building Docker image
BuildKit would speed up the builds, and I strongly doubt we'd run into any of its limitations. I don't think we intend on building any Windows images.

We have two pretty good options:
1) Set env var `DOCKER_BUILDKIT=1`
2) Use something like the [docker/build-push-action](https://github.com/docker/build-push-action) action instead of directly running `docker build`

### PR DESCRIPTION
**Pull request by blakemorgan**
I revisited this and did some maintenance. Resolves #94 and resolves #97. Also addressed some of #104. 

- Update dependabot
- Update Terraform version
- update actions
- Add buildkit
- Update Terraform
- Update tf state and versions

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,8 +5,9 @@ on:
     branches: [dev, stg, master]
     types: [opened, reopened, synchronize, edited]
 env:
+  DOCKER_BUILDKIT: 1
```
Comment (GaryGSC) [inline .github/workflows/ci.yml:8]:
```suggestion
```

I don't think we need this if we're using the new buildx actions.
>Buildx builds using the BuildKit engine and does not require DOCKER_BUILDKIT=1 environment variable to start the builds.

([docs](https://docs.docker.com/buildx/working-with-buildx/#build-with-buildx))

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -4,8 +4,9 @@ on:
   push:
     branches: [dev, stg, master]
 env:
+  DOCKER_BUILDKIT: 1
```
Comment (GaryGSC) [inline .github/workflows/deploy.yml:7]:
```suggestion
```

**Review by yoshutch — APPROVED** [review]

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  # GitHub Actions
+  - package-ecosystem: "github-actions"
+    directory: "/" # For GitHub Actions, set the directory to / to check for workflow files in .github/workflows. (GitHub Docs)
+    schedule:
+      interval: "daily"
+    target-branch: "dev" # Our dev pipeline runs off the dev branch, production runs off of master
+    # Requesting reviews from yourself makes Dependabot PRs easy to find (https://github.com/pulls/review-requested)
+#    reviewers:
+#      - "YourGitHubUsername"
```
Comment (GaryGSC) [inline .github/dependabot.yml:22]: I see we're losing the `ignored_updates` thing. No big deal.

**Comment by GaryGSC** [comment]
@blakemorgan Y'all are quick with these merges.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (GaryGSC) [reply .github/workflows/ci.yml:8]: Closed in #119.

**Pull request merged**